### PR TITLE
improve support for configurations splat in multiple files

### PR DESCRIPTION
**Pull request by benoitdm-oslandia**
This PR follows #567 and fixes review issues.

> I bring a fix to improve the support of splat configurations (via base).
> 
> This fix allows to have heterogeneous yaml files (one with only services, another with one layer definition, another with common configuration parts). It improves the support for None sub dictionaries and can merge lists (like layers).

Ref https://github.com/mapproxy/mapproxy/issues/293

### CODE REVIEW

**Comment by benoitdm-oslandia** [comment]
@olt can you review this PR please?

**Comment by shimoncohen** [comment]
@weskamm Can you please review this PR when you have the time? :)

**Comment by weskamm** [comment]
I cannot say for sure that the comments from the old thread / PR have been tackled here.
Maybe someone else can do a review here? @justb4 @drnextgis @walkermatt @ramunasd @tomkralidis

**Comment by benoitdm-oslandia** [comment]
@justb4 @drnextgis @walkermatt @ramunasd @tomkralidis gentle ping

**Comment by ramunasd** [comment]
This looks interesting. Can You describe this feature with examples in mapproxy docs? @benoitdm-oslandia

**Comment by benoitdm-oslandia** [comment]
@ramunasd I will asap!

**Comment by benoitdm-oslandia** [comment]
rebased with upstream !

File: mapproxy/config/loader.py [hunk 1]
```diff
@@ -2327,12 +2327,55 @@ def merge_dict(conf, base):
...
+                        base[k] = v
+                    elif k in ['layers']:
+                        base[k] = merge_layers(v, base[k])
+                    elif len(v) == 0:  # delete
+                        base[k] = None
+                    else:
+                        base[k] = base[k] + v
             else:
+                # if k in ['layers']:
+                #     # todo: merge list of layer enumeration?
```
Comment (ramunasd) [inline mapproxy/config/loader.py:2344]: Is it possible to resolve this TODO now?

**Comment by benoitdm-oslandia** [comment]
rebase and fix flake error

File: mapproxy/config/loader.py (hunk 1 above)
Comment (ramunasd) [reply mapproxy/config/loader.py:2341]: What do You mean by "if this make sense"?
If this path is not possible then there is no need to leave TODO
If path is possible but there is no implementation, then please throw error respectively.

Comment (benoitdm-oslandia) [reply mapproxy/config/loader.py]: I have dug further and I found no case where this could be needed. I revert, rebase and pushed.

**Comment by benoitdm-oslandia** [comment]
@ramunasd does it look good to you?

**Comment by ramunasd** [comment]
@weskamm I think this one is ready for merge

**Comment by benoitdm-oslandia** [comment]
resolve conflicts in documentation

**Comment by weskamm** [comment]
Thanks for your contribution!

**Comment by benoitdm-oslandia** [comment]
:champagne: !! Thanks @weskamm

**Pull request merged**